Mathlib/LinearAlgebra/Multilinear/Basic.lean

```diff
@@ -1194,6 +1211,330 @@ theorem map_sub [DecidableEq ι] (m : ∀ i, M₁ i) (i : ι) (x y : M₁ i) :
   rw [sub_eq_add_neg, sub_eq_add_neg, MultilinearMap.map_add, map_neg]
 #align multilinear_map.map_sub MultilinearMap.map_sub
 
+/-- This calculates the differences between the values of a multilinear map at
+two arguments that differ on a finset `s` of `ι` of cardinality `n`. It requires a
+linear order on `ι` in order to choose an order isomorphism between `s` and `Fin n`
+and expresses the result using this order isomorphism.-/
+lemma map_sub_finset [LinearOrder ι] (f : MultilinearMap R M₁ M₂) (a b v : (i : ι) → (M₁ i))
+    {s : Finset ι} (hs : Finset.card s = n) :
+    f (s.piecewise a v) - f (s.piecewise b v) = Finset.sum Finset.univ (fun (i : Fin n) =>
+    f (fun j => if h : j ∈ s then (if (s.orderIsoOfFin hs).symm ⟨j, h⟩ < i then a j
+    else if (s.orderIsoOfFin hs).symm ⟨j, h⟩ = i then a j - b j else b j) else v j)) := by
```

Comment:
I think this lemma is till true if you replace `s.orderIsoOfFin hs` with `σ : Fin (s.card) ≃ s` (or perhaps keep `n` and use  `σ : Fin n ≃ s`) throughout, and will be a bit easier to read as a result.

Reply:
This was superseded by alreadydone's simplification.